common/src/main/java/com/microsoft/identity/common/internal/request/SdkType.java

```diff
@@ -22,10 +22,31 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.request;
 
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+
 /**
  * Enum to indicate if the request came from an app hosting Adal or Msal Sdk
  */
 public enum SdkType {
     ADAL,
-    MSAL
-}
\ No newline at end of file
+    MSAL,
+    MSALCPP;
+
+
+    public String getProductName() {
+        if ((SdkType.ADAL == this) || (SdkType.MSAL == this)) {
+            return AuthenticationConstants.SdkPlatformFields.PRODUCT_NAME_MSAL;
+        } else {
+            return AuthenticationConstants.SdkPlatformFields.PRODUCT_NAME_MSAL_CPP;
+        }
+
+    }
+
+    /**
+     * The check for SdkType.MSALCPP is added to ensure
+     * the uniformity of logic with SdkType.MSAL
```

Comment:
Needs a better javadoc

Reply:
Something along the lines of `Determines if the Sdk supports Microsoft Personal Accounts.`

Reply:
any specific suggestions for the change?
How about something on lines of needed check to ensure the request is for a MSA account.

Reply:
addressed